java/ql/src/Security/CWE/CWE-918/RequestForgery.java

```diff
@@ -0,0 +1,20 @@
+import java.net.http.HttpClient;
+
+public class SSRF extends HttpServlet {
+	private static final String VALID_URI = "http://lgtm.com";
+	private HttpClient client = HttpClient.newHttpClient();
+
+	protected void doGet(HttpServletRequest request, HttpServletResponse response)
+	throws ServletException, IOException {
+		URI uri = new URI(request.getParameter("uri"));
+		// BAD: a request parameter is incorporated without validation into a Http request
+        HttpRequest r = HttpRequest.newBuilder(uri).build();
+		client.send(r, null);
+
+		// GOOD: the request parameter is validated against a known fixed string
+		if (VALID_URI.equals(request.getParameter("uri"))) {
```

Comment:
I feel like I'm missing something. Where in the query do you have that logic like this guards against SSRF?

Reply:
I don't have them as of now. it a false positive which we can tackle later if there are too many instances of this type. 
The snippet here is to be included in the qhelp as a correct way of preventing SSRF that's it.